clang/test/Driver/sycl-mlong-double.cpp

```diff
@@ -5,11 +5,27 @@
 // CHECK: clang{{.*}} "-triple" "x86_64-unknown-linux-gnu"
 // CHECK-SAME:  "-mlong-double-64"
 
-/// -mlong-double-128 and -mlong-double-80 are not supported for spir64.
+/// -mlong-double-128 and -mlong-double-80 are not supported for spir64, or SYCL GPU targets.
 // RUN: not %clangxx -c -fsycl -mlong-double-128 -target x86_64-unknown-linux-gnu %s -### 2>&1 | FileCheck --check-prefix=CHECK-128 %s
 // CHECK-128: error: unsupported option '-mlong-double-128' for target 'spir64-unknown-unknown'
 // CHECK-128-NOT: clang{{.*}} "-triple" "-spir64-unknown-unknown" {{.*}} "-mlong-double-128"
 
 // RUN: not %clangxx -c -fsycl -mlong-double-80 -target x86_64-unknown-linux-gnu %s -### 2>&1 | FileCheck --check-prefix=CHECK-80 %s
 // CHECK-80: error: unsupported option '-mlong-double-80' for target 'spir64-unknown-unknown'
 // CHECK-80-NOT: clang{{.*}} "-triple" "-spir64-unknown-unknown" {{.*}} "-mlong-double-80"
+
+// RUN: not %clangxx -c -fsycl -mlong-double-128 -target amd_gpu_gfx1031 %s -### 2>&1 | FileCheck --check-prefix=CHECK-128-AMD %s
+// CHECK-128-AMD: error: unsupported option '-mlong-double-128' for target 'amd_gpu_gfx1031'
+// CHECK-128-AMD-NOT: clang{{.*}} "-triple" "-amd_gpu_gfx1031" {{.*}} "-mlong-double-128"
+
+// RUN: not %clangxx -c -fsycl -mlong-double-80 -target amd_gpu_gfx1031 %s -### 2>&1 | FileCheck --check-prefix=CHECK-80-AMD %s
+// CHECK-80-AMD: error: unsupported option '-mlong-double-80' for target 'amd_gpu_gfx1031'
+// CHECK-80-AMD-NOT: clang{{.*}} "-triple" "-amd_gpu_gfx1031" {{.*}} "-mlong-double-80"
```

Comment:
For these, I believe the expected option syntax is to use `--target=x86_64-unknown-linux-gnu -fsycl-targets=amd_gpu_gfx1031`.